### PR TITLE
Fix duplicate metrics collector registration attempted error

### PR DESCRIPTION
**Pull request by liggitt**
Fixes "duplicate metrics collector registration attempted" error seen in https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/batch/pull-kubernetes-unit/43931/

### CODE REVIEW

**Comment by liggitt** [comment]
Just saw https://github.com/kubernetes/kubernetes/pull/49779, though I think a counter is simpler

**Comment by liggitt** [comment]
cc @deads2k

**Comment by deads2k** [comment]
> Just saw #49779, though I think a counter is simpler

I worry that counters will encourage people to assume "53 is always my replication controller reflector".  The appearance of stable numbers is bad.  Same reason that golang randomize map order traversal intentionally.

**Comment by liggitt** [comment]
unit failed on https://github.com/kubernetes/kubernetes/issues/49958

**Comment by liggitt** [comment]
updated to start at an unstable value

File: staging/src/k8s.io/client-go/tools/cache/reflector.go [hunk 1]
```diff
@@ -98,12 +99,17 @@ func NewReflector(lw ListerWatcher, expectedType interface{}, store Store, resyn
 	return NewNamedReflector(getDefaultReflectorName(internalPackages...), lw, expectedType, store, resyncPeriod)
 }
 
+// reflectorDisambiguator is used to disambiguate started reflectors.
+// initialized to an unstable value to ensure meaning isn't attributed to the suffix.
+var reflectorDisambiguator = int64(time.Now().UnixNano() % 12345)
```
Comment (deads2k) [inline staging/src/k8s.io/client-go/tools/cache/reflector.go]: thanks

Comment (DirectXMan12) [inline staging/src/k8s.io/client-go/tools/cache/reflector.go]: so, I'm kind of uncomfortable depending on time here.  While incredibly unlikely now, it's theoretically feasible to have this called at the same timestamp.

**Review by DirectXMan12 — APPROVED** [review]

**Pull request merged**